File: backend/src/main/java/com/rms/domain/Meal.java
package com.rms.domain;

import com.rms.domain.enums.Status;
import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.Instant;
import java.util.ArrayList;
import java.util.HashSet;
import java.util.List;
import java.util.Set;

@Entity
@Table(name = "meals")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Meal {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String name;

    @Column(name = "image_url", columnDefinition = "TEXT")
    private String imageUrl;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "category_id", nullable = false)
    private Category category;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    @Builder.Default
    private Status status = Status.ACTIVE;

    @Column(columnDefinition = "TEXT")
    private String description;

    @OneToMany(mappedBy = "meal", cascade = CascadeType.ALL, orphanRemoval = true, fetch = FetchType.LAZY)
    @Builder.Default
    private List<MealVariation> variations = new ArrayList<>();

    @ManyToMany(fetch = FetchType.LAZY)
    @JoinTable(
            name = "meal_modifiers",
            joinColumns = @JoinColumn(name = "meal_id"),
            inverseJoinColumns = @JoinColumn(name = "modifier_id")
    )
    @Builder.Default
    private Set<Modifier> modifiers = new HashSet<>();

    @CreationTimestamp
    @Column(name = "created_at", nullable = false, updatable = false)
    private Instant createdAt;

    @UpdateTimestamp
    @Column(name = "updated_at", nullable = false)
    private Instant updatedAt;

    public void setVariations(List<MealVariation> variations) {
        this.variations.clear();
        if (variations != null) {
            variations.forEach(v -> {
                v.setMeal(this);
                this.variations.add(v);
            });
        }
    }

    public void setModifiers(Set<Modifier> modifiers) {
        this.modifiers = modifiers == null ? new HashSet<>() : modifiers;
    }
}

File: frontend/src/types/meal.ts
export interface MealVariation {
  id: number
  name: string
  price: number
}

export interface Meal {
  id: number
  name: string
  description: string
  imageUrl: string | null
  categoryId: number
  status: string
  basePrice: number
  totalPrice: number
  variations: MealVariation[]
  modifierIds: number[]
  createdAt: string
  updatedAt: string
}

export interface MealVariationInput {
  name: string
  price: number
  status: string
}

export interface CreateMealRequest {
  name: string
  imageUrl?: string
  categoryId: number
  status: string
  description?: string
  variations: MealVariationInput[]
  modifierIds: number[]
}

export interface PageResponse<T> {
  content: T[]
  totalElements: number
  totalPages: number
  number: number
  size: number
  first: boolean
  last: boolean
}
